sound/soc/sof/ipc4-topology.c

```diff
@@ -1195,6 +1226,27 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 
 		pipe_widget = swidget->spipe->pipe_widget;
 		pipeline = pipe_widget->private;
+
+		if (pipeline->use_chain_dma) {
+			u32 host_dma_id;
+			u32 fifo_size;
+
+			host_dma_id = platform_params->stream_tag - 1;
+			pipeline->msg.primary |= SOF_IPC4_GLB_CHAIN_DMA_HOST_ID(host_dma_id);
+
+			/* Set SCS bit for S16_LE format only */
+			if (params_format(fe_params) == SNDRV_PCM_FORMAT_S16_LE)
+				pipeline->msg.primary |= SOF_IPC4_GLB_CHAIN_DMA_SCS_MASK;
```

Comment:
`msg.primary |= pipeline->msg.primary;` is used in sof_ipc4_chain_dma_send_msg. What if some pipelines are S16_LE format and others are S32_LE format?

Reply:
In chained dma case there there should be only two pipelines. One at host dai and another at hda end. Since there is no processing involved, the format must be the same for the pipelines.

Reply:
@bardliao pipelines exist only for the sake of tplg-driver compatibility. We do no use them in the DSP at all so the pipeline settings are meaningless for chained DMA.